package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,432 @@
+                self._weights_ker = (np.asarray(manifold_density,
+                                     dtype=np.float64) /
+                                     np.mean(manifold_density))
+
+    def run(self):
+        # will only run if distance matrix not already calculated
+        self._dist_matrix.run()
+        self._scaled_matrix = self._dist_matrix.dist_matrix / self._epsilon
+        # this should be a reference to the same object as
+        # self.dist_matrix.dist_matrix
+        # take negative exponent of scaled matrix to create Isotropic kernel
+        self._kernel = np.exp(-self._scaled_matrix)
+        # define an anistropic diffusion term q
+        q_vector = np.zeros((self._nframes, ))
+        # weights should reflect the density of the points on the manifold
+        for i in range(self._nframes):
+            q_vector[i] = np.dot(self._kernel[i, :], self._weights_ker)
+
+        # Form a new kernel from the anisotropic diffusion term q
+        self._kernel /= np.sqrt(q_vector[:, np.newaxis].dot(q_vector[np.newaxis]))
+
+        # Weighted Graph Laplacian normalization on this new graph
+        d_vector = np.zeros((self._nframes, ))
+        for i in range(self._nframes):
+            d_vector[i] = np.dot(self._kernel[i, :], self._weights_ker)
+
+        for i in range(self._nframes):
+            self._kernel[i, :] = self._kernel[i, :] * self._weights_ker
+
+        # Define anisotropic transition by dividing kernel by this term
+        self._kernel /= np.sqrt(d_vector[:, np.newaxis].dot(d_vector[np.newaxis]))
+
+        # Apply timescaling
+        for i in range(self._t):
+            if i > 0:
+                self._kernel = self._kernel.dot(self._kernel)
+
+        eigenvals, eigenvectors = np.linalg.eig(self._kernel)
+        eg_arg = np.argsort(eigenvals)
+        self.eigenvalues = eigenvals[eg_arg[::-1]]
+        self.eigenvalues = self.eigenvalues[1:]
+        self.eigenvectors = eigenvectors[eg_arg[::-1], :]
+        self.eigenvectors = self.eigenvectors[1:]
+        self._calculated = True
+
+    def transform(self, num_eigenvectors):
+        """ Embeds a trajectory via the diffusion map
+
+        Parameter
+        ---------
+        num_eigenvectors : int
+            The number of dominant eigenvectors to be used for
+            diffusion mapping
+
+        Return
+        ------
+        diffusion_space : array
+            The diffusion map embedding as defined by [Ferguson1]_.
+            This isn't a linear transformation, but an isometry
+            between the higher dimensional space and the space spanned by
+            the eigenvectors.
+        """
+        self.diffusion_space = np.zeros((self.eigenvectors.shape[0],
+                                         num_eigenvectors))
+
+        for i in range(self._nframes-1):
+            for j in range(num_eigenvectors):
+                self.diffusion_space[i][j] = self.eigenvectors[j][i]
```

Comment:
This is just a transpose operation. `self.diffusionspace = self.eigenvectors.T[:, :num_eigenvectors]`.
Another issues is that I think you aren't scaling the diffusion space correctly. According to the literature you have to multiply the eigenvector with the eigenvalues to get the "projection" into the diffusion space.